purekit/src/main/kotlin/com/virgilsecurity/purekit/pure/PureStorage.java

```diff
@@ -33,95 +33,172 @@
+     * @throws ProtocolHttpException Thrown if an error from the PHE service has NOT been parsed
+     * successfully. Represents a regular HTTP exception with code and message.
      */
-    void insertUser(UserRecord userRecord) throws Exception;
+    void insertUser(UserRecord userRecord)
+        throws ProtocolException, ProtocolHttpException;
 
     /**
-     * Updates user in storage
-     * @param userRecord User record
-     * @throws Exception FIXME
+     * Updates a user in a storage.
+     *
+     * @param userRecord User record.
+     *
+     * @throws ProtocolException Thrown if an error from the PHE service has been parsed
+     * successfully.
+     * @throws ProtocolHttpException Thrown if an error from the PHE service has NOT been parsed
+     * successfully. Represents a regular HTTP exception with code and message.
      */
-    void updateUser(UserRecord userRecord) throws Exception;
+    void updateUser(UserRecord userRecord)
+        throws ProtocolException, ProtocolHttpException;
 
     /**
-     * Obtains user record with given userId from storage
-     * @param userId userId
-     * @return UserRecord
-     * @throws Exception FIXME
+     * Obtains a user record with the given userId from a storage.
+     *
+     * @param userId User Id.
+     *
+     * @return UserRecord.
+     *
+     * @throws PureException If a user has not been found in a storage or user id mismatches the one
+     * from a server. Use {@link PureException#getErrorStatus()} to know the particular case.
+     * @throws ProtocolException Thrown if an error from the PHE service has been parsed
+     * successfully.
+     * @throws ProtocolHttpException Thrown if an error from the PHE service has NOT been parsed
+     * successfully. Represents a regular HTTP exception with code and message.
      */
-    UserRecord selectUser(String userId) throws Exception;
+    UserRecord selectUser(String userId)
+        throws PureException, ProtocolException, ProtocolHttpException;
 
     /**
-     * Obtains users record with given userId from storage
-     * @param userIds userIds
-     * @return UserRecords
-     * @throws Exception FIXME
+     * Obtains a users record with the given userId from a storage.
+     *
+     * @param userIds User Ids. Should not contain duplicates.
+     *
+     * @return UserRecords.
+     *
+     * @throws PureException If user Id duplicate has been found or user id mismatches the one
+     * from a server. Use {@link PureException#getErrorStatus()} to know the particular case.
+     * @throws ProtocolException Thrown if an error from the PHE service has been parsed
+     * successfully.
+     * @throws ProtocolHttpException Thrown if an error from the PHE service has NOT been parsed
+     * successfully. Represents a regular HTTP exception with code and message.
      */
-    Iterable<UserRecord> selectUsers(Collection<String> userIds) throws Exception;
+    Collection<UserRecord> selectUsers(Collection<String> userIds)
```

Comment:
What's the reason of this change?

Reply:
Map is Iterable as well. Are we ok with passing Map here, for example?

Reply:
Interface should be as less restrictive as possible. AFAIK usages of this interface don't need anything Iterable doesn't have